src/main/java/jenkins/metrics/api/Metrics.java

```diff
@@ -480,6 +488,7 @@ private void execute(TaskListener listener) throws IOException, InterruptedExcep
             for (Map.Entry<String, HealthCheck.Result> e : results.entrySet()) {
                 count++;
                 listener.getLogger().println(" * " + e.getKey() + ": " + e.getValue());
+                healthCheckResults.put(e.getKey(), e.getValue());
```

Comment:
Need to clear out any removed keys... e.g. `heathCheckResults.retainAll(results.keySet())`